src/controllerInput.cpp

```diff
@@ -0,0 +1,76 @@
+#include "controllerInput.hpp"
+
+#include "logger.hpp"
+
+InputState ControllerInput::getInputState(SDL_Keycode key) {
+	// TODO Move SDL_Keycode s to InputKey
+	return NONE;
+}
+
+InputState ControllerInput::getInputState(InputKey key) {
+	auto iterator = mInputStates.find(key);
+	if (iterator == mInputStates.end()) {
+		return NONE;
+	}
+
+	return iterator->second;
+}
+
+void ControllerInput::preUpdateInput() {
+	// XXX: I hate that I have to do this. For some reason game controller events are not being received. Instead of trying to receive them,
+	// since there is only a handful amount of buttons, we check each button instead. And instead of checking events we literally ask about
+	// every single button........ BAH
```

Comment:
I think this is fine. Especially when we add more features where the user is pressing multiple keys at once (e.g shooting, ducking, moving at the same time) this will still work.

Reply:
Yeah, it's probably fine. I just like the idea of looking at the events more instead of force checking everything.